src/main/java/org/mule/extension/ftp/internal/connection/FtpConnectionProvider.java

```diff
@@ -257,21 +257,30 @@ private ConnectionException handleClientReplyCode(int replyCode) {
    * @return A {@link FTPConnectionException} specifying the error cause with a {@link FileError}
    */
   protected ConnectionException handleClientReplyCode(int replyCode, Throwable cause) {
+    FTPConnectionException exception;
     switch (replyCode) {
       case 501:
       case 530:
-        return new FTPConnectionException(getErrorMessage(replyCode, "User cannot log in"),
-                                          INVALID_CREDENTIALS);
+        exception = new FTPConnectionException(getErrorMessage(replyCode, "User cannot log in"),
+                                               INVALID_CREDENTIALS);
+        break;
       case 421:
-        return new FTPConnectionException(getErrorMessage(replyCode, "Service is unavailable"),
-                                          SERVICE_NOT_AVAILABLE);
-    }
-    if (cause != null) {
-      return new FTPConnectionException(getErrorMessage(connectionSettings, format("Error code: '%d'", replyCode)), cause,
-                                        CONNECTIVITY);
+        exception = new FTPConnectionException(getErrorMessage(replyCode, "Service is unavailable"),
+                                               SERVICE_NOT_AVAILABLE);
+        break;
+      default:
+        if (cause != null) {
+          exception =
+              new FTPConnectionException(getErrorMessage(connectionSettings, format("Error code: '%d'", replyCode)), cause,
+                                         CONNECTIVITY);
+        } else {
+          exception = new FTPConnectionException(getErrorMessage(connectionSettings, format("Error code: '%d'", replyCode)));
+        }
     }
 
-    return new FTPConnectionException(getErrorMessage(connectionSettings, format("Error code: '%d'", replyCode)));
+    LOGGER.error(exception.getMessage(), exception);
+
```

Comment:
I would leave this as it was, and make it the function caller's choice to log the error